core/src/vaults.ts

```diff
@@ -0,0 +1,361 @@
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _fetchGlobalLiquidationLimits = async (network: string) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const maximumProtocolDebtDaiHex = await contract.Hole();
+    const maximumProtocolDebtDai = new BigNumber(maximumProtocolDebtDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const currentProtocolDebtDaiHex = await contract.Dirt();
+    const currentProtocolDebtDai = new BigNumber(currentProtocolDebtDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    return {
+        currentProtocolDebtDai,
+        maximumProtocolDebtDai,
+    };
+};
+
+export const fetchGlobalLiquidationLimits = memoizee(_fetchGlobalLiquidationLimits, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 1,
+});
+
+const _fetchCollateralLiquidationLimitsAndLiquidatorAddress = async (
+    network: string,
+    collateralType: CollateralType
+) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const contractVat = await getContract(network, 'MCD_VAT');
+    const typeHex = ethers.utils.formatBytes32String(collateralType);
+    const { hole, dirt, clip, chop } = await contract.ilks(typeHex);
+    const { dust } = await contractVat.ilks(typeHex);
+    return {
+        currentCollateralDebtDai: new BigNumber(dirt._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS),
+        maximumCollateralDebtDai: new BigNumber(hole._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS),
+        liquidatiorContractAddress: clip,
+        liquidationPenaltyRatio: new BigNumber(chop._hex).shiftedBy(-WAD_NUMBER_OF_DIGITS),
+        minimalAuctionedDai: new BigNumber(dust._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS),
+    };
+};
+
+export const fetchCollateralLiquidationLimitsAndLiquidatorAddress = memoizee(
+    _fetchCollateralLiquidationLimitsAndLiquidatorAddress,
+    {
+        maxAge: CACHE_EXPIRY_MS,
+        promise: true,
+        length: 2,
+    }
+);
+
+const _fetchVault = async (network: string, index: number): Promise<Vault> => {
+    const vaultBase = await fetchVaultBase(network, index);
+    const vaultCollateralParameters = await fetchVaultCollateralParameters(network, vaultBase.collateralType);
+    const vaultAmount = await fetchVaultAmount(network, vaultBase.collateralType, vaultBase.address);
+    const globalLiquidationLimits = await fetchGlobalLiquidationLimits(network);
+    const { currentCollateralDebtDai, maximumCollateralDebtDai, liquidationPenaltyRatio, minimalAuctionedDai } =
+        await fetchCollateralLiquidationLimitsAndLiquidatorAddress(network, vaultBase.collateralType);
+    return {
+        ...vaultBase,
+        ...vaultCollateralParameters,
+        ...vaultAmount,
+        ...globalLiquidationLimits,
+        currentCollateralDebtDai,
+        maximumCollateralDebtDai,
+        liquidationPenaltyRatio,
+        minimalAuctionedDai,
+    };
+};
+export const fetchVault = memoizee(_fetchVault, {
+    maxAge: CACHE_EXPIRY_MS,
```

Comment:
I don't think this should ever be cached

Reply:
is this outdated? i struggle to find this line 🥇